src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -130,19 +136,24 @@ export default class PhotoField extends React.Component {
               this.props.onChange({
                 errorMessage: 'The photo you uploaded isn’t a square photo. Please upload a new one that fits the requirements.'
               });
+              this.setState({ currentLayout: layouts.screenReaderError });
             } else if (!isValidImageSize(img)) {
               this.props.onChange({
                 errorMessage: 'The file you selected is smaller than the 350-pixel minimum file width or height and couldn’t be uploaded. Please try to upload a different photo.'
               });
+
+              this.setState({ currentLayout: layouts.screenReaderError });
             } else {
-              this.setState({ progress: 0, warningMessage: null });
+              this.setState({
+                currentLayout: layouts.watchUpload,
+                progress: 0,
+                warningMessage: null
+              });
               this.uploadRequest = this.props.formContext.uploadFile(
                 file,
                 this.props.uiSchema['ui:options'],
                 this.updateProgress,
-                (fileData) => {
-                  this.props.onChange(fileData);
-                },
+                (formData) => this.handleUploadComplete(formData, file),
```

Comment:
Screenreader handler for upload complete got out of sync with the `onChange` handler. Want to make sure that both of them should handle the upload complete the same way- I noticed an issue where the screen reader upload wasn't showing the preview image after uploading it.

Reply:
Yeah, they should be the same.